MergeSharpWebAPI/Program.cs

```diff
@@ -55,28 +55,32 @@ private static void StartServer()
         app.Run();
     }
 
-    private static async Task Main(string[] args)
+    private static void connectionReconnected()
     {
-        Thread serverThread = new Thread(StartServer);
-        serverThread.Start();
-
-        connection.Reconnecting += error =>
+        connection.Reconnected += async connectionId =>
         {
-            System.Diagnostics.Debug.Assert(connection.State == HubConnectionState.Reconnecting);
-
-            // Notify users the connection was lost and the client is reconnecting.
-            // Start queuing or dropping messages.
-
-            return Task.CompletedTask;
+            if (connection.State == HubConnectionState.Connected)
+            {
+                Console.WriteLine("RECONNECTED");
+                await Task.Delay(new Random().Next(0, 5) * 1000);
```

Comment:
why do we have to wait some time and why is the amount of time random between 0 and 5s?

Reply:
Tru @canlasla do u know the reason? I noticed it's already implemented like that in main, so I think we copied it from the SignalR Hubs example, but Idk what its purpose is either

Reply:
If two clients are on the same machine, then they both go offline by turning on airplane mode, then they both come back online, its to allow one client to send the encoded message first to the server before the other client by delaying by a random amount of time. 

If this is not there, both clients send the encoded message at the same time but which ever client is quicker sends there's first, leaving the other clients encoded message to be lost

Reply:
ah ok

it would be good to add a comment explaining this. Comments should describe anything non obvious that its clear from just reading the code